components/cam/src/dynamics/se/restart_dynamics.F90

```diff
@@ -389,20 +460,42 @@ subroutine read_restart_dynamics (File, dyn_in, dyn_out, NLFileName)
     call PIO_InitDecomp(pio_subsystem, pio_double, (/ncols,nlev/),ldof , iodesc3d)
     deallocate(ldof)
 
+#ifdef MODEL_THETA_L
+    if ( .not. theta_hydrostatic_mode ) then
+       allocate(var3dp(s2d*nlevp))
+    endif
+    var3dp = 0.0
+
```

Comment:
`var3dp` appears to be unallocated when `theta_hydrostatic_mode=.true.`:
```
  4:
  4: Program received signal SIGSEGV: Segmentation fault - invalid memory reference.
  4:
  4: Backtrace for this error:
  4: #0  0x2b94e6fc227f in ???
  4: #1  0xd3ffa6 in __restart_dynamics_MOD_read_restart_dynamics
  4:    at /lcrc/group/acme/azamat/repos/E3SM-bebop-master/components/cam/src/dynamics/se/restart_dynamics.F90:467
  4: #2  0x567c28 in __cam_restart_MOD_cam_read_restart
  4:    at /lcrc/group/acme/azamat/repos/E3SM-bebop-master/components/cam/src/control/cam_restart.F90:397
  4: #3  0x5185ca in __cam_comp_MOD_cam_init
  4:    at /lcrc/group/acme/azamat/repos/E3SM-bebop-master/components/cam/src/control/cam_comp.F90:167
  4: #4  0x510b62 in __atm_comp_mct_MOD_atm_init_mct
  4:    at /lcrc/group/acme/azamat/repos/E3SM-bebop-master/components/cam/src/cpl/atm_comp_mct.F90:314
  4: #5  0x433174 in __component_mod_MOD_component_init_cc
  4:    at /lcrc/group/acme/azamat/repos/E3SM-bebop-master/cime/src/drivers/mct/main/component_mod.F90:248
  4: #6  0x421bc7 in __cime_comp_mod_MOD_cime_init
  4:    at /lcrc/group/acme/azamat/repos/E3SM-bebop-master/cime/src/drivers/mct/main/cime_comp_mod.F90:1337
  4: #7  0x42f5b9 in cime_driver
  4:    at /lcrc/group/acme/azamat/repos/E3SM-bebop-master/cime/src/drivers/mct/main/cime_driver.F90:122
  4: #8  0x42f73a in main
  4:    at /lcrc/group/acme/azamat/repos/E3SM-bebop-master/cime/src/drivers/mct/main/cime_driver.F90:23
```
This is during restart run of `ERS.ne4_ne4.FC5AV1C-L.anvil-centos7_gnu.cam-thetahy_ftype2` on latest master.

Reply:
Thanks! I will push a fix. Would it be caught with DEBUG?

Reply:
gcc-8.2.0 is running into this in both debug and non-debug mode. I'm not sure if intel will catch this in debug mode. Nightlies are in non-debug mode.

Reply:
Ok, out-of-curiosity, Intel v.17.0.4 in debug mode reports a different error, later in the restart run:
```
 91: forrtl: error (65): floating invalid
 91: Image              PC                Routine            Line        Source
 91: e3sm.exe           0000000009ACF132  Unknown               Unknown  Unknown
 91: libpthread-2.17.s  00002B657BA545D0  Unknown               Unknown  Unknown
 91: e3sm.exe           0000000003208753  prim_state_mod_mp         230  prim_state_mod.F90
 91: e3sm.exe           0000000002FC8D2D  prim_driver_base_         949  prim_driver_base.F90
 91: e3sm.exe           00000000041E9AFE  dyn_comp_mp_dyn_i         322  dyn_comp.F90
 91: e3sm.exe           0000000002CEB34D  restart_dynamics_         681  restart_dynamics.F90
 91: e3sm.exe           00000000009EF860  cam_restart_mp_ca         397  cam_restart.F90
 91: e3sm.exe           000000000084AFA5  cam_comp_mp_cam_i         167  cam_comp.F90
 91: e3sm.exe           00000000008164E0  atm_comp_mct_mp_a         312  atm_comp_mct.F90
 91: e3sm.exe           0000000000465390  component_mod_mp_         257  component_mod.F90
 91: e3sm.exe           000000000042A509  cime_comp_mod_mp_        1337  cime_comp_mod.F90
 91: e3sm.exe           000000000045C119  MAIN__                    122  cime_driver.F90
```
No errors in non-debug mode.

Reply:
Line 230 in prim_state_mod is 
`       psmin_local(ie) = MINVAL(tmp(:,:,ie))`
where 
`       tmp(:,:,ie)=hvcoord%hyai(1)*hvcoord%ps0 + sum(elem(ie)%state%dp3d(:,:,:,n0),3)`
I would assume there are nans in dp3d from some other issue. How do you run? Is it still a restart test? I would like to try.

Reply:
Same test. This was on Bebop `ERS_D.ne4_ne4.FC5AV1C-L.bebop_intel.cam-thetahy_ftype2`.

Reply:
I am running on anvil right now. can I use bebop? I could ssh to it.

Reply:
Yes, anyone with access to Anvil should be able to run on Bebop after `ssh bebop.lcrc.anl.gov`.

Reply:
I see the same on anvil. I will debug it. Meanwhile, could we please have `srun -K` for anvil? my job got stuck.

Reply:
@amametjanov  see https://github.com/E3SM-Project/E3SM/pull/3426 , finally debugged this . thanks!

Reply:
That is referring to ERS_D.ne4_ne4.FC5AV1C-L.anvil_gnu.cam-thetahy_ftype2. I thought the recent error was about anvil+intel (not gnu).

Reply:
GNU run was also crashing in printstate mode. I ran with intel just now, all is ok:
```
(base) [onguba@blueslogin4 as]$ ./cs.status.20200204_200805_z29jlq
20200204_200805_z29jlq: 1 test
  ERS_D.ne4_ne4.FC5AV1C-L.anvil_intel.cam-thetahy_ftype2 (Overall: PASS) details:
    PASS ERS_D.ne4_ne4.FC5AV1C-L.anvil_intel.cam-thetahy_ftype2 CREATE_NEWCASE
    PASS ERS_D.ne4_ne4.FC5AV1C-L.anvil_intel.cam-thetahy_ftype2 XML
    PASS ERS_D.ne4_ne4.FC5AV1C-L.anvil_intel.cam-thetahy_ftype2 SETUP
    PASS ERS_D.ne4_ne4.FC5AV1C-L.anvil_intel.cam-thetahy_ftype2 SHAREDLIB_BUILD time=154
    PASS ERS_D.ne4_ne4.FC5AV1C-L.anvil_intel.cam-thetahy_ftype2 MODEL_BUILD time=203
    PASS ERS_D.ne4_ne4.FC5AV1C-L.anvil_intel.cam-thetahy_ftype2 SUBMIT
    PASS ERS_D.ne4_ne4.FC5AV1C-L.anvil_intel.cam-thetahy_ftype2 RUN time=708
    PASS ERS_D.ne4_ne4.FC5AV1C-L.anvil_intel.cam-thetahy_ftype2 COMPARE_base_rest
    PASS ERS_D.ne4_ne4.FC5AV1C-L.anvil_intel.cam-thetahy_ftype2 MEMLEAK
    PASS ERS_D.ne4_ne4.FC5AV1C-L.anvil_intel.cam-thetahy_ftype2 SHORT_TERM_ARCHIVER
```